src/Tools/dotnet-counters/Exporters/ConsoleWriter.cs

```diff
@@ -150,31 +150,43 @@ public void AssignRowsAndInitializeDisplay()
             {
                 Console.WriteLine($"[{provider.Name}]"); row++;
 
+                foreach (var counter in provider.Counters.Values)
+                {
+                    counter.Row = -1;
+                    if (counter.RenderValueInline)
+                    {
+                        foreach (var tagSet in counter.TagSets.Values)
+                        {
+                            tagSet.Row = -1;
+                        }
+                    }
+                }
+
                 foreach (ObservedCounter counter in provider.Counters.Values.OrderBy(c => c.DisplayName))
                 {
 
                     string name = MakeFixedWidth($"{new string(' ', Indent)}{counter.DisplayName}", Indent + _maxNameLength);
+                    if (row >= _consoleHeight + _topRow - 1) // prevents from displaying more counters than vertical space available
+                    {
+                        break;
+                    }
                     counter.Row = row++;
                     if (counter.RenderValueInline)
                     {
-                        if (row >= _consoleHeight) // prevents from displaying more counters than vertical space available
-                        {
-                            break;
-                        }
                         Console.WriteLine($"{name} {FormatValue(counter.LastValue)}");
                     }
                     else
                     {
                         Console.WriteLine(name);
                         foreach (ObservedTagSet tagSet in counter.TagSets.Values.OrderBy(t => t.Tags))
                         {
-                            if (row >= _consoleHeight)
+                            string tagName = MakeFixedWidth($"{new string(' ', 2 * Indent)}{tagSet.Tags}", Indent + _maxNameLength);
+                            Console.WriteLine($"{tagName} {FormatValue(tagSet.LastValue)}");
+
```

Comment:
Why did you switch the order of the Console.WriteLine and the check to make sure the row number is not greater than or equal to the Console height?

Reply:
Great question.  The order is switched so that we don't assign the counter a row number if we don't end up rendering it.  The logic is as follows:
* When a new counter is found via the tracevent callback, we reset all of the counter object row numbers to -1 and then recompute them.
* Once the row number scrolls off the end of the screen (by comparing the console height and row number) we terminate the loop.  This means all counter objects that can be rendered have a row number >= 0, and all counter objects that are not visible have a row number of -1.